Exercise_2.java

```diff
@@ -9,30 +9,56 @@
         StackNode(int data) 
         { 
             //Constructor here 
+            this.data = data;
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
         //Write your code here for the condition if stack is empty. 
+        if(root==null)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
     } 
   
     public void push(int data) 
     { 
         //Write code to push data to the stack. 
+        StackNode oldroot = root;
+        root = new StackNode(data);
+        root.next = oldroot;
+
+
     } 
   
     public int pop() 
     { 	
 	//If Stack Empty Return 0 and print "Stack Underflow"
-        //Write code to pop the topmost element of stack.
-	//Also return the popped element 
+    //Write code to pop the topmost element of stack.
+    //Also return the popped element 
+        if(isEmpty())
+        {
+            System.out.println("Stack Underflow");
+            return 0;
+        }
+        else
+        {
+            int popedItem = root.data;
+            root = root.next;
+            return popedItem;
+        }
     } 
   
     public int peek() 

```

Comment:
when the stack is empty and I call the peek function, I should be notified that stack is empty by either printing a default value or error message 
but in this case, I get a null pointer exception, rectify it